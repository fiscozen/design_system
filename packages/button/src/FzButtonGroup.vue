<template>
  <div :class="computedClasses">
    <slot></slot>
  </div>
</template>

<script lang="ts" setup>
import { computed } from 'vue'
import type { ButtonSize } from './types'

const props = withDefaults(
  defineProps<{
    /**
     * The direction of the button group. If true, buttons will be displayed horizontally, otherwise vertically
     */
    horizontal?: boolean
    /**
     * Whether to add a gap between buttons
     */
    gap?: boolean
    /**
     * The gap size between buttons
     */
    size?: ButtonSize
  }>(),
  {
    horizontal: true,
    size: 'md'
  }
)

const computedClasses = computed(() => {
  return {
    'flex flex-col vertical': !props.horizontal,
    'flex flex-row horizontal': props.horizontal,
    'gap-disabled': !props.gap,
    'gap-8': props.gap && props.size === 'xs',
    'gap-10': props.gap && props.size === 'sm',
    'gap-12': props.gap && props.size === 'md',
    'gap-16': props.gap && props.size === 'lg'
  }
})
</script>

<style scoped>
.gap-disabled :deep(button:not(:first-child):not(:last-child)) {
  border-radius: 0;
}

.gap-disabled.horizontal :deep(button:first-child) {
  border-top-right-radius: 0;
  border-bottom-right-radius: 0;
}

.gap-disabled.horizontal :deep(button:last-child) {
  border-top-left-radius: 0;
  border-bottom-left-radius: 0;
}

.gap-disabled.horizontal :deep(button:not(:first-child)) {
  margin-left: -1px;
}

.gap-disabled.vertical :deep(button:first-child) {
  border-bottom-left-radius: 0;
  border-bottom-right-radius: 0;
}

.gap-disabled.vertical :deep(button:last-child) {
  border-top-left-radius: 0;
  border-top-right-radius: 0;
}

.gap-disabled.vertical :deep(button:not(:first-child)) {
  margin-top: -1px;
}
</style>
