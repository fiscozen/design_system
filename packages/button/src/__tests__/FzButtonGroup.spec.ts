// FzButtonGroup.spec.ts

import { mount } from '@vue/test-utils'
import { expect, describe, it } from 'vitest'
import FzButtonGroup from '../FzButtonGroup.vue'

describe('FzButtonGroup', () => {
  it('renders correctly', () => {
    const wrapper = mount(FzButtonGroup, {
      props: {
        gap: true,
        size: 'md'
      },
      slots: {
        default: '<button>Button 1</button><button>Button 2</button>'
      }
    })
    expect(wrapper.html()).to.include('Button 1')
    expect(wrapper.html()).to.include('Button 2')
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('applies correct gap class based on size prop', () => {
    const wrapper = mount(FzButtonGroup, {
      props: {
        gap: true,
        size: 'md'
      }
    })
    expect(wrapper.classes()).to.include('gap-12')
  })

  it('does not apply gap class when gap prop is false', () => {
    const wrapper = mount(FzButtonGroup, {
      props: {
        gap: false,
        size: 'md'
      }
    })
    expect(wrapper.classes()).to.include('gap-disabled')
  })

  it("applies 'vertical' class when horizontal prop is false", () => {
    const wrapper = mount(FzButtonGroup, {
      props: {
        gap: false,
        size: 'md',
        horizontal: false
      }
    })
    expect(wrapper.classes()).to.include('flex-col')
  })
})
