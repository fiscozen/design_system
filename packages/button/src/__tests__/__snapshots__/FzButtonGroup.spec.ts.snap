// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FzButtonGroup > renders correctly 1`] = `"<div data-v-830cd86f="" class="flex flex-row horizontal gap-12"><button>Button 1</button><button>Button 2</button></div>"`;
